addons/mail/static/src/js/form_renderer.js

```diff
@@ -1,77 +1,151 @@
 odoo.define('mail.form_renderer', function (require) {
 "use strict";
 
-var Chatter = require('mail.Chatter');
-var FormRenderer = require('web.FormRenderer');
+const Chatter = require('mail.component.Chatter');
+const FormRenderer = require('web.FormRenderer');
 
 /**
  * Include the FormRenderer to instanciate the chatter area containing (a
  * subset of) the mail widgets (mail_thread, mail_followers and mail_activity).
  */
 FormRenderer.include({
+    async on_attach_callback() {
+        this._super(...arguments);
+        // When the view is re-attached (_isInDom is true) and the chatter component already exists
+        // the chatter component needs to be retargeted in the DOM by force mounting.
+        if (this._chatterComponent && this._isInDom) {
+            await this._forceMountChatterComponent();
+        }
+    },
+    on_detach_callback() {
+        this._super(...arguments);
+        // When the view is detached, we totally delete chatter state from store and chatter
+        // component to avoid any problem when view will be reattached
+        if (this._chatterComponent) {
+            this._deleteChatter();
+        }
+    },
     /**
      * @override
      */
-    init: function (parent, state, params) {
-        this._super.apply(this, arguments);
+    init(parent, state, params) {
+        this._super(...arguments);
+        this.env = this.call('messaging', 'getMessagingEnv');
         this.mailFields = params.mailFields;
-        this.chatter = undefined;
+        // Do not load chatter in form view dialogs
+        this._preventChatterLoad = params.fromFormViewDialog;
+        this._chatterComponent = undefined;
+        this._chatterLocalId = undefined;
+        /**
+         * Determine if the form view has a chatter
+         * (i.e. if a div .oe-chatter is detected in the arch)
+         * @type {boolean}
+         * @private
+         */
+        this._hasChatter = false;
```

Comment:
Maybe remove this attribute, just keep `_chatterContainerTarget` and mount chatter only when this is truthy.

Reply:
I found it clearer but I agree it's redundant, I removed it

Reply:
Looks a bit clearer, but I think drawback of syncing this attribute with `_chatterContainerTarget` outweighs the gain.